### PR TITLE
#18182 missing remove attribute when encapsulating method

### PR DESCRIPTION
**Pull request by erickgonzalez**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java [hunk 1]
```diff
@@ -403,7 +403,7 @@ private void updateLoginAsSessionInfo(final HttpServletRequest request, final Ho
 		final String userToImpersonate = (UtilMethods.isSet(loginAsUserId) ? loginAsUserId : principalUserId);
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java]: should this change have a postman test or at least a unit/integration test?

**Review by freddyucv — CHANGES_REQUESTED** [review]

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java [hunk 2]
```diff
@@ -498,6 +498,7 @@ public final Response logoutAs(@Context final HttpServletRequest httpServletRequ
 			final Map<String, Object> sessionData = this.helper.doLogoutAs(principalUserId, currentLoginAsUser, serverName);
 			revertLoginAsSessionInfo(httpServletRequest, Host.class.cast(sessionData.get(com.dotmarketing.util.WebKeys
 					.CURRENT_HOST)), principalUserId);
+			httpServletRequest.getSession().removeAttribute(WebKeys.PRINCIPAL_USER_ID);
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java]: Move this code to the `revertLoginAsSessionInfo` method in order to encapsulate the session revert process

**Review by jcastro-dotcms — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java (hunk 2 above)
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java]: done

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java (hunk 1 above)
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java:403]: done

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    static HttpServletResponse response;
+    static HttpServletRequest request;
+    static UserResource resource;
+    static User user;
+    static Host host;
+    static User adminUser;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:35]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        assertNotNull(resourceResponse);
+        assertEquals(resourceResponse.getEntity().toString(), Status.OK.getStatusCode(),resourceResponse.getStatus());
+        assertEquals(user.getUserId(),request.getSession().getAttribute(WebKeys.USER_ID));
+        assertNull(request.getSession().getAttribute(WebKeys.USER));
+        assertEquals(adminUser.getUserId(),request.getSession().getAttribute(WebKeys.PRINCIPAL_USER_ID));
+        assertEquals(host,request.getSession().getAttribute(com.dotmarketing.util.WebKeys.CURRENT_HOST));
+    }
+
+    @Test
+    public void test_loginAs_success() throws Exception{
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:86]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+        request.setHeader("Authorization",
+                "Basic " + new String(Base64.encode("admin@dotcms.com:admin".getBytes())));
+
+        request.setAttribute(com.dotmarketing.util.WebKeys.CURRENT_HOST,host);
+
+        return request;
+    }
+
+    private void loginAs() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:73]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.WebKeys;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.glassfish.jersey.internal.util.Base64;
+import static org.junit.Assert.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:21]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:35]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:86]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:73]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:21]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+    static HttpServletResponse response;
+    static HttpServletRequest request;
+    static UserResource resource;
+    static User user;
+    static Host host;
+    static User adminUser;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:35]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        request.setHeader("Authorization",
+                "Basic " + new String(Base64.encode("admin@dotcms.com:admin".getBytes())));
+
+        request.getSession().setAttribute(com.dotmarketing.util.WebKeys.CURRENT_HOST,host);
+        request.getSession().setAttribute(com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID,host);
+
+        return request;
+    }
+
+    private void loginAs() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:74]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+        assertNotNull(resourceResponse);
+        assertEquals(resourceResponse.getEntity().toString(), Status.OK.getStatusCode(),resourceResponse.getStatus());
+        assertEquals(user.getUserId(),request.getSession().getAttribute(WebKeys.USER_ID));
+        assertNull(request.getSession().getAttribute(WebKeys.USER));
+        assertEquals(adminUser.getUserId(),request.getSession().getAttribute(WebKeys.PRINCIPAL_USER_ID));
+        assertEquals(host,request.getSession().getAttribute(com.dotmarketing.util.WebKeys.CURRENT_HOST));
+    }
+
+    @Test
+    public void test_loginAs_success() throws Exception{
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:87]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.WebKeys;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.glassfish.jersey.internal.util.Base64;
+import static org.junit.Assert.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:21]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:87]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:35]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java [hunk 11]
```diff
@@ -440,19 +448,26 @@ private void revertLoginAsSessionInfo(final HttpServletRequest request, final Ho
...
 		final HttpSession session = req.getSession();
 		final User user = this.userAPI.loadUserById(userID);
+		Logger.info(this,"USER: " + user);
 		final String currentSiteID = (String) session.getAttribute(com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID);
+		Logger.info(this,"currentSiteID: " + currentSiteID);
 		Host currentSite;
 		try {
 			currentSite = this.siteAPI.find(currentSiteID, user, false);
 		} catch (final DotSecurityException e) {
+			Logger.info(this,"currentSite: " + "DOTSECURITY");
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java:458]: Codacy Issue found: [The String literal "currentSite: " appears 5 times in this file; the first occurrence is on line 458](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:74]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:21]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+        assertNotNull(resourceResponse);
+        assertEquals(resourceResponse.getEntity().toString(), Status.OK.getStatusCode(),resourceResponse.getStatus());
+        assertEquals(user.getUserId(),request.getSession().getAttribute(WebKeys.USER_ID));
+        assertNull(request.getSession().getAttribute(WebKeys.USER));
+        assertEquals(adminUser.getUserId(),request.getSession().getAttribute(WebKeys.PRINCIPAL_USER_ID));
+        assertEquals(host,request.getSession().getAttribute(com.dotmarketing.util.WebKeys.CURRENT_HOST));
+    }
+
+    @Test
+    public void test_loginAs_success() throws Exception{
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:90]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+import com.dotmarketing.business.PermissionAPI;
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.WebKeys;
+import io.vavr.API;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.glassfish.jersey.internal.util.Base64;
+import static org.junit.Assert.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:22]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java:458]: Codacy Issue found: [The String literal "currentSite: " appears 5 times in this file; the first occurrence is on line 458](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+        request.setHeader("Authorization",
+                "Basic " + new String(Base64.encode("admin@dotcms.com:admin".getBytes())));
+
+        request.getSession().setAttribute(com.dotmarketing.util.WebKeys.CURRENT_HOST,host);
+        request.getSession().setAttribute(com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID,host);
+
+        return request;
+    }
+
+    private void loginAs() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:77]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+import com.dotcms.mock.request.MockSessionRequest;
+import com.dotcms.mock.response.MockHttpResponse;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Permission;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.WebKeys;
+import io.vavr.API;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:16]: Codacy Issue found: [Avoid unused imports such as 'io.vavr.API'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+
+    static HttpServletResponse response;
+    static HttpServletRequest request;
+    static UserResource resource;
+    static User user;
+    static Host host;
+    static User adminUser;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:36]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+import com.dotcms.mock.request.MockSessionRequest;
+import com.dotcms.mock.response.MockHttpResponse;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Permission;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.WebKeys;
+import io.vavr.API;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:18]: Codacy Issue found: [Avoid unused imports such as 'io.vavr.API'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.glassfish.jersey.internal.util.Base64;
+import static org.junit.Assert.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:24]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java:458]: Codacy Issue found: [The String literal "currentSite: " appears 5 times in this file; the first occurrence is on line 458](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+        final Response resourceResponse = resource.loginAs(request,response,loginAsForm);
+        assertNotNull(resourceResponse);
+        assertEquals(resourceResponse.getEntity().toString(), Status.OK.getStatusCode(),resourceResponse.getStatus());
+        assertEquals(user.getUserId(),request.getSession().getAttribute(WebKeys.USER_ID));
+        assertNull(request.getSession().getAttribute(WebKeys.USER));
+        assertEquals(adminUser.getUserId(),request.getSession().getAttribute(WebKeys.PRINCIPAL_USER_ID));
+    }
+
+    @Test
+    public void test_loginAs_success() throws Exception{
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:91]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 20]
```diff
@@ -0,0 +1,107 @@
+package com.dotcms.rest.api.v1.user;
+
+import com.dotcms.datagen.SiteDataGen;
+import com.dotcms.datagen.TestDataUtils;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:4]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen.TestDataUtils'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 21]
```diff
@@ -0,0 +1,107 @@
...
+
+    static HttpServletResponse response;
+    static HttpServletRequest request;
+    static UserResource resource;
+    static User user;
+    static Host host;
+    static User adminUser;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:38]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+        request.setHeader("Authorization",
+                "Basic " + new String(Base64.encode("admin@dotcms.com:admin".getBytes())));
+
+        request.getSession().setAttribute(com.dotmarketing.util.WebKeys.CURRENT_HOST,host);
+        request.getSession().setAttribute(com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID,host);
+
+        return request;
+    }
+
+    private void loginAs() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:79]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 20 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:4]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen.TestDataUtils'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java:458]: Codacy Issue found: [The String literal "currentSite: " appears 5 times in this file; the first occurrence is on line 458](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 22 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:79]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 17 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:18]: Codacy Issue found: [Avoid unused imports such as 'io.vavr.API'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 18 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:24]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 21 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:38]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java (hunk 19 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:91]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 23]
```diff
@@ -0,0 +1,103 @@
...
+
+    static HttpServletResponse response;
+    static HttpServletRequest request;
+    static UserResource resource;
+    static User user;
+    static Host host;
+    static User adminUser;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:36]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 24]
```diff
@@ -0,0 +1,103 @@
...
+        final Response resourceResponse = resource.loginAs(request,response,loginAsForm);
+        assertNotNull(resourceResponse);
+        assertEquals(Status.OK.getStatusCode(),resourceResponse.getStatus());
+        assertEquals(user.getUserId(),request.getSession().getAttribute(WebKeys.USER_ID));
+        assertNull(request.getSession().getAttribute(WebKeys.USER));
+        assertEquals(adminUser.getUserId(),request.getSession().getAttribute(WebKeys.PRINCIPAL_USER_ID));
+    }
+
+    @Test
+    public void test_loginAs_success() throws Exception{
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:87]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 25]
```diff
@@ -0,0 +1,103 @@
...
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.WebKeys;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.glassfish.jersey.internal.util.Base64;
+import static org.junit.Assert.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:22]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java [hunk 26]
```diff
@@ -0,0 +1,103 @@
...
+        request.setHeader("Authorization",
+                "Basic " + new String(Base64.encode("admin@dotcms.com:admin".getBytes())));
+
+        request.getSession().setAttribute(com.dotmarketing.util.WebKeys.CURRENT_HOST,host);
+        request.getSession().setAttribute(com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID,host);
+
+        return request;
+    }
+
+    private void loginAs() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java:75]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)

**Pull request merged**